sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosVectorIndexSpec.java

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.CosmosItemSerializer;
+import com.azure.cosmos.implementation.Constants;
+import com.azure.cosmos.implementation.JsonSerializable;
+
+/**
+ * Vector Indexes spec for Azure CosmosDB service.
+ */
+public final class CosmosVectorIndexSpec {
+
+    private JsonSerializable jsonSerializable;
+    private String type;
+
+    /**
+     * Constructor
+     */
+    public CosmosVectorIndexSpec() { this.jsonSerializable = new JsonSerializable(); }
+
+    /**
+     * Gets path.
+     *
+     * @return the path.
+     */
+    public String getPath() {
+        return this.jsonSerializable.getString(Constants.Properties.PATH);
+    }
+
+    /**
+     * Sets path.
+     *
+     * @param path the path.
+     * @return the SpatialSpec.
+     */
+    public CosmosVectorIndexSpec setPath(String path) {
+        this.jsonSerializable.set(Constants.Properties.PATH, path, CosmosItemSerializer.DEFAULT_SERIALIZER);
+        return this;
+    }
+
+    /**
+     * Gets the vector index type for the vector index
+     *
+     * @return the vector index type
+     */
+    public String getType() {
+        if (this.type == null) {
+            this.type = this.jsonSerializable.getString(Constants.Properties.VECTOR_INDEX_TYPE);
+
+            if (this.type == null) {
+                throw new IllegalArgumentException("INVALID vectorIndexType of " + this.jsonSerializable.getString(Constants.Properties.VECTOR_INDEX_TYPE));
```

Comment:
this is very uncommon to throw in getter for null value, perhaps the null check and throwing should be moved to the type setter?